src/components/ToggleSwitch.js

```diff
@@ -39,12 +40,17 @@ class ToggleSwitch extends React.Component {
     this.props.onToggle(event);
   };
 
-  render () {
+  _handleKeyDown = (event) => {
+    event.preventDefault();
+    if (isEnterOrSpaceKey(event)) this._handleToggle(event);
+  };
+
+  render() {
     const theme = StyleUtils.mergeTheme(this.props.theme);
     const styles = this.styles(theme);
 
     return (
-      <div className='toggle-switch-component' ref={this.props.elementRef} style={styles.component}>
+      <div aria-checked={this.props.checked} className='toggle-switch-component' onKeyDown={this._handleKeyDown} ref={this.props.elementRef} role="switch" style={styles.component} tabIndex={0} >
```

Comment:
If we don't provide the `switch` role, the screen reader will treat it as just as a regular button and read off "button". If we apply the role the SR will be enhanced and read it as a switch like "on, switch" or "off, switch".

I believe it would be appropriate to have this role applied.

Reply:
Sounds good to me :)